client/src/test/java/io/pravega/client/connection/impl/RawClientTest.java

```diff
@@ -78,6 +79,24 @@ public void testRequestReply() throws InterruptedException, ExecutionException,
         assertEquals(reply, future.get());
     }
 
+    @Test
+    public void testRecvErrorMessage() throws InterruptedException, ExecutionException, ConnectionFailedException {
+        PravegaNodeUri endpoint = new PravegaNodeUri("localhost", -1);
+        @Cleanup
+        MockConnectionFactoryImpl connectionFactory = new MockConnectionFactoryImpl();
+        @Cleanup
+        MockController controller = new MockController(endpoint.getEndpoint(), endpoint.getPort(), connectionFactory, true);
+        ClientConnection connection = Mockito.mock(ClientConnection.class);
+        connectionFactory.provideConnection(endpoint, connection);
+        Segment segment = new Segment("scope", "testHello", 0);
+        @Cleanup
+        RawClient rawClient = new RawClient(controller, connectionFactory, segment);
```

Comment:
This is not used. A request should be sent and the reply returned should be verified.

Reply:
Instantiation the `RawClient` is required as it sets up the connection to provided the `ReplyProcessor`. I chose not to actually send a request because of the complications of implicitly relying on segmentstore logic to throw back the expected error.